Library/Homebrew/cmd/info.rb

```diff
@@ -123,8 +123,12 @@ def info_formula(f)
     puts f.desc if f.desc
     puts Formatter.url(f.homepage) if f.homepage
 
-    conflicts = f.conflicts.map(&:name).sort!
-    puts "Conflicts with: #{conflicts*", "}" unless conflicts.empty?
+    conflicts = f.conflicts.map do |c|
+      c.name +
+        (c.reason ? " (because #{c.reason})" : "")
+    end.sort!
+    msg="Conflicts with: "
+    puts msg+conflicts*(",\n"+" "*msg.length) unless conflicts.empty?
```

Comment:
I'm not sure how to do as you request and make the alignment across multiple lines work right. By setting `msg` first, the code as written can use the length of that line to pad the continuation lines with 13 blanks so it lines up properly, but without hardcoding the count. I'm not sure how to do what you ask without losing that. Maybe I want to be too fancy?